pkg/controller/controller.go

```diff
@@ -146,6 +147,15 @@ func NewManagerForRestConfig(conf *config.Config, rc *rest.Config) (ctrl.Manager
 	return m, nil
 }
 
+// IsPrometheusBestPracticeName - function returns true if the name given matches best practices for prometheus name, i.e. snake_case
+func IsPrometheusBestPracticeName(controllerName string) bool {
+	pattern := "[a-z]+(_[a-z]+)*"
+	re := regexp.MustCompile(pattern)
+	match := re.FindStringSubmatch(controllerName)
```

Comment:
This would be a bit cleaner if we use

https://pkg.go.dev/regexp#Regexp.MatchString

`MatchString` and control for the entire string with ^ and $ regex.